Apply Gas refactor

Daksh14 opened the pull request:
- Update CHANGELOG.md
- Update README.md with new information

Closes #70

Daksh14 committed:
- Apply Gas refactor
- Update CHANGELOG.md
- Update README.md with new information

ZER0 commented on src/mod.js:
```diff
@@ -8,14 +8,34 @@ import { getPsks } from "./keys.js";
+  // Passing null/undefined/0 or negative values will set the default value for price and limit
+  constructor({ limit, price } = {}) {
+    this.limit = Math.max(limit, 0) || 2_900_000_000;
+    this.price = Math.max(price, 0) || 1;
```
Let's add the two default as static of the class. Something like:
```js
export class Gas {
  static DEFAULT_LIMIT = 2_900_000_000;
  static DEFAULT_PRICE = 1;

  limit = NaN;
  price = NaN;

  constructor({ limit, price }  = {}) {
    this.limit = Math.max(limit, 0) || Gas.DEFAULT_LIMIT;
    this.price = Math.max(price, 0) || Gas.DEFAULT_PRICE;
    
    Object.freeze(this);
  }
}
```

ZER0 commented on src/mod.js:
```diff
@@ -69,15 +88,15 @@ export class Wallet {
    * @returns {Promise} promise that resolves after the transfer is accepted into blockchain
    */
-  transfer(sender, receiver, amount) {
+  transfer(sender, receiver, amount, gas = new Gas()) {
```
Shouldn't the docs also reflect the new optional argument?

ZER0 commented on src/mod.js:
```diff
@@ -87,7 +106,7 @@ export class Wallet {
    * @returns {Promise} promise that resolves after the stake is accepted into blockchain
    */
-  async stake(staker, amount) {
+  async stake(staker, amount, gas = new Gas()) {
```
Same here

ZER0 commented on src/mod.js:
```diff
@@ -148,38 +167,25 @@ export class Wallet {
    * @returns {Promise} promise that resolves after the unstake is accepted into blockchain
    */
-  async unstake(unstaker) {
+  async unstake(unstaker, gas = new Gas()) {
```
And here.

ZER0 commented on src/mod.js:
```diff
@@ -148,38 +167,25 @@ export class Wallet {
    * @returns {Promise} promise that resolves after the unstake is accepted into blockchain
    */
-  async withdrawReward(psk) {
+  async withdrawReward(psk, gas = new Gas()) {
```
And here too.

ZER0 commented on tests/balance_test.js:
```diff
@@ -250,3 +250,34 @@ Deno.test({
+    assertEquals(gas.limit, 2_900_000_000);
+  },
+  sanitizeResources: false,
+  sanitizeOps: false,
```
You can remove those two, as mentioned in the comment above in the first tests you need those only if you're handling indexedDB (and you are not), since the fake version of it it's faulty. 

Also, I would add a test for checking the limit when you pass a correct value, and that you can easily "clone" a new instance of `Gas` class. Something like:

```js
    gas = new Gas({ price: 3, limit: 1_200_000_000});
    let gas2 = new Gas(gas);
    
    assertEquals(gas2.price, 3);
    assertEquals(gas2.limit, 1_2000_000_000);
```

ZER0 reviewed (changes requested):
Looks good but there are few small nits: let's add them and do another round of review.

Daksh14 committed:
Add `DEFAULT_LIMIT` and `DEFAULT_PRICE` static vars
Add jsdoc param for newly added `gas` argument
Add gas limit override Deno test

ZER0 reviewed (approved):
LGTM
